dm/syncer/binlogstream/streamer_controller.go

```diff
@@ -115,14 +77,42 @@ func (r *remoteBinlogReader) GenerateStreamer(location binlog.Location) (reader.
 	}
 
 	// position's name may contain uuid, so need remove it
-	adjustedPos := binlog.AdjustPosition(location.Position)
+	adjustedPos := binlog.RemoveRelaySubDirSuffix(location.Position)
 	streamer, err := r.reader.StartSync(adjustedPos)
 	return streamer, terror.ErrSyncerUnitRemoteSteamerStartSync.Delegate(err)
 }
 
+type locationedStream struct {
```

Comment:
```suggestion
type locationStream struct {
```
location can only be a noun.

Reply:
I want to express this stream has a "location" property :joy: Can you give a better name?

Reply:
`LocationStream` is informative enough, isn't it?